Closes #20834: Add support for enabling/disabling Tokens

pheus opened the pull request:
### Fixes: #20834

This PR introduces an `enabled` field on the `Token` model to allow API tokens to be temporarily disabled without being deleted.

Key changes:

- Add an `enabled` Boolean field to `Token` (defaulting to `True` for backwards compatibility)
- Enforce the `enabled` flag in token authentication so disabled tokens are rejected
- Expose the new field in token forms, bulk edit, and tables to make it easy to toggle status from the UI
- Expose `enabled` via the token API serializers (including provisioning), allowing tokens to be created and managed in a disabled state
- Add tests for the Token model, API, and authentication to cover the new behavior

Existing tokens remain enabled by default, so there is no change in behavior until a token is explicitly disabled.

Thanks in advance for reviewing!

pheus commented:
I’ve just rebased this PR onto the newly merged branch and resolved the merge conflicts. The changes are now up to date and ready for review.

Thank you very much for taking a look!

jeremystretch commented on netbox/users/models/tokens.py:
```diff
@@ -180,6 +185,26 @@ def token(self, value):
+            return False
+        if self.is_expired:
+            return False
+        return True
```
We can simplify this a bit.
```suggestion
        return self.enabled and not self.is_expired
```

pheus replied:
💯 
This is way better! Thank you!

jeremystretch reviewed (commented):
Nice work @pheus! Just had a few minor suggestions.

pheus committed:
feat(users): Add support for enabling/disabling Tokens

Introduce an `enabled` flag on the `Token` model to allow temporarily
revoking API tokens without deleting them. Update forms, serializers,
and views to expose the new field.
Enforce the `enabled` flag in token authentication.
Add model, API, and authentication tests for the new behavior.

Fixes #20834

pheus commented on netbox/users/tests/test_models.py:
```diff
@@ -20,6 +20,26 @@ def setUpTestData(cls):
+        self.assertFalse(token.is_active)
+
+        # Token with disabled status
+        token.enabled = False
```
Agreed. I've added all possible combination in the updated commit. Thank you!

pheus commented:
Thank you very much for the thoughtful review and suggestions! I really appreciate you taking the time to go through this. Your feedback is super helpful and will definitely help me improve!

jeremystretch committed:
Fix authentication test
